compiler/rustc_session/src/search_paths.rs

```diff
@@ -124,14 +124,19 @@ impl SearchPath {
             early_dcx.early_fatal("empty search path given via `-L`");
         }
 
+        if !dir.exists() {
+            #[allow(rustc::untranslatable_diagnostic)] // FIXME: make this translatable
+            early_dcx.early_warn(format!("search path `{}` does not exist", dir.display()));
```

Comment:
I think we‘d better to explicitly inform users that this search path comes from the `-L` option.